### PR TITLE
typos: init at 1.1.9

### PR DESCRIPTION
**Pull request by mgttlinger**
###### Motivation for this change

package typos a source code spell checker.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/typos/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    rev = "v${version}";
+    sha256 = "sha256:10ydsp77v4kf1qsq5wyc02iyfsdy0rpcyxycp2lqz9qy3g3ih7vm";
+  };
+
+  cargoSha256 = "sha256:1i6999nmg4pahpp4fz4qm4rx8iixa13zjwlhyixwjwbag1w8l3gp";
+
+  meta = with lib; {
+    description = "Source code spell checker";
+    homepage = "https://https://github.com/crate-ci/typos/";
+    license = [ licenses.asl20 licenses.mit ]; # dual licensed
```
Comment (SuperSandro2000) [inline pkgs/development/tools/typos/default.nix:19]:
```suggestion
    license = with licenses; [ asl20 /* or */ mit ];
```

File: pkgs/development/tools/typos/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256:10ydsp77v4kf1qsq5wyc02iyfsdy0rpcyxycp2lqz9qy3g3ih7vm";
+  };
+
+  cargoSha256 = "sha256:1i6999nmg4pahpp4fz4qm4rx8iixa13zjwlhyixwjwbag1w8l3gp";
+
+  meta = with lib; {
+    description = "Source code spell checker";
+    homepage = "https://https://github.com/crate-ci/typos/";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/typos/default.nix:18]:
```suggestion
    homepage = "https://github.com/crate-ci/typos/";
```

File: pkgs/development/tools/typos/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  version = "1.1.9";
+
+  src = fetchFromGitHub {
+    owner = "crate-ci";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256:10ydsp77v4kf1qsq5wyc02iyfsdy0rpcyxycp2lqz9qy3g3ih7vm";
+  };
+
+  cargoSha256 = "sha256:1i6999nmg4pahpp4fz4qm4rx8iixa13zjwlhyixwjwbag1w8l3gp";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/typos/default.nix:14]:
```suggestion
    sha256 = "10ydsp77v4kf1qsq5wyc02iyfsdy0rpcyxycp2lqz9qy3g3ih7vm";
  };

  cargoSha256 = "1i6999nmg4pahpp4fz4qm4rx8iixa13zjwlhyixwjwbag1w8l3gp";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by mgttlinger** [comment]
Thank you for the suggestions @SuperSandro2000!

**Pull request merged**